packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -227,16 +333,51 @@ class _BottomNavigationTile extends StatelessWidget {
     // produce smooth animation. We do this by multiplying the flex value
     // (which is an integer) by a large number.
     int size;
-    Widget label;
+
+    // Defines the padding for the animating icons + labels.
+    //
+    // The animations go from "Unselected":
+    // =======
+    // |      <-- Padding equal to the text height.
+    // |  ☆
+    // | text <-- Invisible text.
+    // =======
+    //
+    // To "Selected":
+    //
+    // =======
+    // |      <-- Padding equal to 1/2 text height.
+    // |  ☆
+    // | text
+    // |      <-- Padding equal to 1/2 text height.
+    // =======
+    double bottomPadding = Tween<double>(
```

Comment:
It seems like we're doing a bit of extra work here. Could we re-arrange this code to avoid performing the evaluation if one of the conditions below is true?

Reply:
OK, moved so that the default doesn't actually touch the `Tween`s.